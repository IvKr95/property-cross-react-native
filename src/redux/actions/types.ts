export const SET_SEARCHES = 'SET_SEARCHES';
export const REMOVE_SEARCHES = 'REMOVE_SEARCHES';
export const SET_FIELD = 'SET_FIELD';
export const SET_LOCATIONS = 'SET_LOCATIONS';
export const SET_PROPS = 'SET_PROPS';
export const REMOVE_PROPS = 'REMOVE_PROPS';
export const SET_ERROR = 'SET_ERROR';
export const SET_LISTING = 'SET_LISTING';
export const SET_FAVOURITE = 'SET_FAVOURITE';
export const SET_FAVOURITES = 'SET_FAVOURITES';
export const REMOVE_FAVOURITE = 'REMOVE_FAVOURITE';
export const REMOVE_LISTING = 'REMOVE_LISTING';
export const SET_LOADING = 'SET_LOADING';
