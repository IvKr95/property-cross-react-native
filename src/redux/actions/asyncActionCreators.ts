import locationAPI from '../../api/locationApi';
import {Dispatch} from 'redux';

interface AsyncActions {
  REQUEST: string;
  SUCCESS: string;
  FAILURE: string;
  ENDED: string;
}

const getAsyncActions = (type: string) => ({
  REQUEST: `${type}_REQUEST`,
  SUCCESS: `${type}_SUCCESS`,
  FAILURE: `${type}_FAILURE`,
  ENDED: `${type}_ENDED`,
});

export const searchLocation: Function = (params: object) => (
  dispatch: Dispatch,
): void => {
  const SEARCH_LOCATION = getAsyncActions('SEARCH_LOCATION');
  dispatch({type: SEARCH_LOCATION.REQUEST});

  locationAPI.getLocation(params).then(
    result => {
      if (result.listings) {
        dispatch({
          type: `${SEARCH_LOCATION.SUCCESS}_LISTINGS`,
          payload: result.listings,
        });
      } else {
        dispatch({
          type: `${SEARCH_LOCATION.SUCCESS}_LOCATIONS`,
          payload: result.locations,
        });
      }
      dispatch({type: SEARCH_LOCATION.ENDED});
    },
    error => {
      dispatch({type: SEARCH_LOCATION.FAILURE, payload: error.message});
      dispatch({type: SEARCH_LOCATION.ENDED});
    },
  );
};
