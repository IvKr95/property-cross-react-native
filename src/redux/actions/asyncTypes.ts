const getAsyncActions = (type: string) => ({
  REQUEST: `${type}_REQUEST`,
  SUCCESS: `${type}_SUCCESS`,
  FAILURE: `${type}_FAILURE`,
});

export const SEARCH_LOCATION = getAsyncActions('SEARCH_LOCATION');
