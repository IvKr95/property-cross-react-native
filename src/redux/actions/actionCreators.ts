import * as types from './types';
import {SEARCH_LOCATION} from './asyncTypes';
import {Action, RecentSearch, Listing} from '../../interfaces';

export const setSearches = (payload: RecentSearch[]): Action => ({
  type: types.SET_RECENT_SEARCHES,
  payload,
});

export const setSearch = (payload: RecentSearch): Action => ({
  type: types.SET_RECENT_SEARCH,
  payload,
});

export const removeSearch = (payload: string): Action => ({
  type: types.REMOVE_RECENT_SEARCH,
  payload,
});

export const setSearchField = (payload: string): Action => ({
  type: types.SET_SEARCH_FIELD,
  payload,
});

export const setListing = (payload: Listing): Action => ({
  type: types.SET_LISTING,
  payload,
});

export const removeListing = (): Action => ({
  type: types.REMOVE_LISTING,
});

export const setFavourite = (payload: Listing): Action => ({
  type: types.SET_FAVOURITE,
  payload,
});

export const setFavourites = (payload: Listing[]): Action => ({
  type: types.SET_FAVOURITES,
  payload,
});

export const removeFavourite = (payload?: string): Action => ({
  type: types.REMOVE_FAVOURITE,
  payload,
});

export const removeFavourites = (): Action => ({
  type: types.REMOVE_FAVOURITES,
});

export const setError = (payload: string): Action => ({
  type: types.SET_ERROR,
  payload,
});

export const searchLocationRequest = (payload: object): Action => ({
  type: SEARCH_LOCATION.REQUEST,
  payload,
});

export const searchLocationSuccess = (payload: object): Action => ({
  type: SEARCH_LOCATION.SUCCESS,
  payload,
});

export const searchLocationFailure = (payload: string): Action => ({
  type: SEARCH_LOCATION.FAILURE,
  payload,
});

export const getGeolocation = (): Action => ({
  type: types.GET_GEOLOCATION,
});
