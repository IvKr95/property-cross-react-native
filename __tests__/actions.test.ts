import * as actionCreators from '../src/redux/actions/actionCreators';

describe('setSearches returns actions', () => {
  it('returns an action with an array of objects', () => {
    expect(actionCreators.setSearches([{id: 'testID'}])).toMatchSnapshot();
  });

  it('returns an action with empty array', () => {
    expect(actionCreators.setSearches([])).toMatchSnapshot();
  });
});

describe('setFavourites returns actions', () => {
  it('returns an action with an array of objects', () => {
    expect(
      actionCreators.setFavourites([{someParam: 'someParam'}]),
    ).toMatchSnapshot();
  });

  it('returns an action with empty array', () => {
    expect(actionCreators.setFavourites([])).toMatchSnapshot();
  });
});

describe('removeFavourite returns an action', () => {
  it('returns an id in payload', () => {
    expect(actionCreators.removeFavourite('someId')).toMatchSnapshot();
  });

  it('returns undefined in payload', () => {
    expect(actionCreators.removeFavourite()).toMatchSnapshot();
  });
});

test('setSearch returns an action', () => {
  expect(actionCreators.setSearch({id: 'testID'})).toMatchSnapshot();
});

test('removeSearch returns an action', () => {
  expect(actionCreators.removeSearch('testId')).toMatchSnapshot();
});

test('setSearchField returns an action', () => {
  expect(actionCreators.setSearchField('testSearch')).toMatchSnapshot();
});

test('setListing returns an action', () => {
  expect(actionCreators.setListing({id: 'someId'})).toMatchSnapshot();
});

test('removeListing returns an action', () => {
  expect(actionCreators.removeListing()).toMatchSnapshot();
});

test('setFavourite returns an action', () => {
  expect(actionCreators.setFavourite({id: 'someId'})).toMatchSnapshot();
});

test('removeFavourites returns an action', () => {
  expect(actionCreators.removeFavourites()).toMatchSnapshot();
});

test('setError returns an action', () => {
  expect(actionCreators.setError('someError')).toMatchSnapshot();
});

test('getGeolocation return an action', () => {
  expect(actionCreators.getGeolocation()).toMatchSnapshot();
});
