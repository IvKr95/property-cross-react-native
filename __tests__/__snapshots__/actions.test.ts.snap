// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getGeolocation return an action 1`] = `
Object {
  "type": "GET_GEOLOCATION",
}
`;

exports[`removeFavourite returns an action returns an id in payload 1`] = `
Object {
  "payload": "someId",
  "type": "REMOVE_FAVOURITE",
}
`;

exports[`removeFavourite returns an action returns undefined in payload 1`] = `
Object {
  "payload": undefined,
  "type": "REMOVE_FAVOURITE",
}
`;

exports[`removeFavourites returns an action 1`] = `
Object {
  "type": "REMOVE_FAVOURITES",
}
`;

exports[`removeListing returns an action 1`] = `
Object {
  "type": "REMOVE_LISTING",
}
`;

exports[`removeSearch returns an action 1`] = `
Object {
  "payload": "testId",
  "type": "REMOVE_RECENT_SEARCH",
}
`;

exports[`setError returns an action 1`] = `
Object {
  "payload": "someError",
  "type": "SET_ERROR",
}
`;

exports[`setFavourite returns an action 1`] = `
Object {
  "payload": Object {
    "id": "someId",
  },
  "type": "SET_FAVOURITE",
}
`;

exports[`setFavourites returns actions returns an action with an array of objects 1`] = `
Object {
  "payload": Array [
    Object {
      "someParam": "someParam",
    },
  ],
  "type": "SET_FAVOURITES",
}
`;

exports[`setFavourites returns actions returns an action with empty array 1`] = `
Object {
  "payload": Array [],
  "type": "SET_FAVOURITES",
}
`;

exports[`setListing returns an action 1`] = `
Object {
  "payload": Object {
    "id": "someId",
  },
  "type": "SET_LISTING",
}
`;

exports[`setSearch returns an action 1`] = `
Object {
  "payload": Object {
    "id": "testID",
  },
  "type": "SET_RECENT_SEARCH",
}
`;

exports[`setSearchField returns an action 1`] = `
Object {
  "payload": "testSearch",
  "type": "SET_SEARCH_FIELD",
}
`;

exports[`setSearches returns actions returns an action with an array of objects 1`] = `
Object {
  "payload": Array [
    Object {
      "id": "testID",
    },
  ],
  "type": "SET_RECENT_SEARCHES",
}
`;

exports[`setSearches returns actions returns an action with empty array 1`] = `
Object {
  "payload": Array [],
  "type": "SET_RECENT_SEARCHES",
}
`;
